### PR TITLE
[#5077] feat(core): Add the check of user before creating metadata object

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?

Add the check of user before creating metadata object.

### Why are the changes needed?

Fix: #5077 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?
Added UT.

### CODE REVIEW

File: clients/client-java/src/test/java/org/apache/gravitino/client/integration/test/authorization/CheckCurrentUserIT.java [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+            "thrift://%s:%d",
+            containerSuite.getHiveContainer().getContainerIpAddress(),
+            HiveContainer.HIVE_METASTORE_PORT);
+
+    containerSuite.startKafkaContainer();
+    kafkaBootstrapServers =
+        String.format(
+            "%s:%d",
+            containerSuite.getKafkaContainer().getContainerIpAddress(),
+            KafkaContainer.DEFAULT_BROKER_PORT);
```
Comment (xunliu) [inline clients/client-java/src/test/java/org/apache/gravitino/client/integration/test/authorization/CheckCurrentUserIT.java:85]: Why do we need to use a Kafka container to test?
If we can use Hive container to test at the same time, we can save resources.

File: clients/client-java/src/test/java/org/apache/gravitino/client/integration/test/authorization/CheckCurrentUserIT.java
Comment (xunliu) [inline clients/client-java/src/test/java/org/apache/gravitino/client/integration/test/authorization/CheckCurrentUserIT.java:1]: Currently, This UT only have throw ForbiddenException UT.
I think we need to add normal operation UT.

File: clients/client-java/src/test/java/org/apache/gravitino/client/integration/test/authorization/CheckCurrentUserIT.java (hunk 1 above)
Comment (jerqi) [reply clients/client-java/src/test/java/org/apache/gravitino/client/integration/test/authorization/CheckCurrentUserIT.java:85]: We need to test the creating topic case.

File: clients/client-java/src/test/java/org/apache/gravitino/client/integration/test/authorization/CheckCurrentUserIT.java
Comment (jerqi) [reply clients/client-java/src/test/java/org/apache/gravitino/client/integration/test/authorization/CheckCurrentUserIT.java:1]: In other cases, we have created the metadata objects. It's ok to add it.

File: core/src/main/java/org/apache/gravitino/hook/CatalogHookDispatcher.java [hunk 2]
```diff
@@ -71,6 +72,10 @@ public Catalog createCatalog(
       String comment,
       Map<String, String> properties)
       throws NoSuchMetalakeException, CatalogAlreadyExistsException {
+    // Check whether the current user exists or not
+    AuthorizationUtils.checkCurrentUser(
```
Comment (jerryshao) [inline core/src/main/java/org/apache/gravitino/hook/CatalogHookDispatcher.java:76]: Do we only need to check `createXXX` operation?

Comment (jerqi) [reply core/src/main/java/org/apache/gravitino/hook/CatalogHookDispatcher.java:76]: We should check every operations actually. 
We don't support meta authentication now. I think this should be included to meta authentication. 
But for some managers, we don't a dispatcher interface for them like ownerManager and TagManager.  We will add the interfaces first. I'm not sure to do them now. So I just make a minor improvement to check `createXXX` operation first.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**